### DISCUSSION TITLE
About how to generate tables in database while running the shaduler

### DISCUSSION THREAD

**Discussion by priya23595** [Sunbird-cQube/community#809] [Q&A]
How are datasets tables in the Postgres Database generated based on events and dimensions CSV. Need to know the flow for it
can u help me to understand

**Comment by DevashishRaghav** [comment]
Hey Priya
Please check this deck: [Link](https://docs.google.com/presentation/d/1b9k6UgmKBEcXrYLtHNEwBM3R5Rj9wmzE2Yt71UXYe64/edit?usp=sharing)

**Reply by bommalla** [reply]
@priya23595 Every grammar file should have one data file then only datasets are created in the database otherwise it won't created.
1. The adaptor API will pick the files from the bucket and create one process_input folder and store the all event files with program name separately  in the process input folder.
2. The "Run Latest Code local" nifi processor take input files + dimensions based on that mapping will happens laters Nifi having certain datasets tables list with queries are executed and storing output to respective dataset table

**Comment by priya23595** [comment]
I will Go through this link, thank you 
But we have some questions can we connect in second half

**Reply by RNarmatha** [reply]
Hi Priya, 
Can you please post your questions here . So that our team can get in touch with you

**Comment by priya23595** [comment]
How to get automatically generate the tables in database while run the scheduler after ingestion the data? what exactly code for this need to understand

**Reply by RNarmatha** [reply]
https://github.com/Sunbird-cQube/processing-ms/tree/staging/impl/c-qube

Please look into this .

**Comment by priya23595** [comment]
can we connect? about this understanding

**Comment by DevashishRaghav** [comment]
Hey Priya
Sure we can connect but before that, we will soon put up the recordings of training sessions on microsite. To get the maximum value from our interaction over a call, I think it would be great if you go through them first.
Thanks

**Comment by priya23595** [comment]
ok sir no problem please share this training videos soon
thank you

**Comment by priya23595** [comment]
Please can u share the link of microsite so that i can go through the training sessions recording videos 
Thank you

**Comment by DevashishRaghav** [comment]
Here is the link: 
https://cqube.sunbird.org/

Under the 'Recordings of Training' section, you will find the link to the Google folder where you can access all recordings.

**Comment by priya23595** [comment]
-Hello can we connect about ingesting and uploading the project elated csv  files on minio bucket and visualise the data
-Please can u help ,me to ingesting the new dimension and event project related files and visualise the same 
can we connect?

**Reply by RNarmatha** [reply]
Hi @priya23595 , For the ingestion , we have the steps in the microsite . Kindly follow the [steps ](https://cqube.sunbird.org/data-ingestion-and-processing/setting-up-postman)

**Comment by priya23595** [comment]
and also want to know about code changes for other data

**Reply by RNarmatha** [reply]
For this can you please elaborate  on this point

**Comment by priya23595** [comment]
-I am having new data for ingestion ,after ingesting the data we need to run scheduler it will run the script file and generating the tables in database and then it will visualise it on dashboard.
-For this new data what exactly code change we need to do in script or about the generating the tables needs to understand the code

**Reply by RNarmatha** [reply]
@dhanush-2397 - Could you please help on this ?

**Reply by dhanush-2397** [reply]
@priya23595 If the data is similiar to the schema and the grammar files present then we don't need any code changes =>  (VSK programs already present in the code base), 
If you are working on your own use case then you need to define event grammars,dimension grammars and then modify the config.json file to ingest the data.

**Reply by priya23595** [reply]
In Which micro-services this config file is present

**Comment by priya23595** [comment]
yes am Working for our own use-cases related data files can u tell me the steps of how to define the event grammar and  dimension grammars?

**Reply by RNarmatha** [reply]
Please refer this [link](https://cqube.sunbird.org/steps-to-prepare-data/define-dimension-schema) to define the event and dimensions

**Comment by priya23595** [comment]
I am not able to delete previously ingested  the event and dimension data from dimension grammar and event grammar tables can u help to delete this data to defining new data

**Reply by bommalla** [reply]
Please follow the below steps to truncate the data.
1. First check the all the containers: sudo docker ps -a
2. Go inside nifi container: sudo docker exec -it nifi_app bash
3. Then navigate the below mentioned path: Sunbird-cQube-processing-ms/impl/c-qube/inest
4. Then run the below command : yarn cli nuke-db

**Comment by priya23595** [comment]
Hello
I Am Done with creating the schema but when am uploading with national program or in new programs then it will get error
![Screenshot from 2023-08-03 16-52-59](https://github.com/Sunbird-cQube/community/assets/136596601/5d9d8677-e74b-48ce-b24c-fe866076e5fc)

**Comment by bommalla** [comment]
@priya23595 Please upload the csv file instead of zip.

**Comment by priya23595** [comment]
Hello 
-Now am able to upload dimension and event csv file using new programs but not seen in minio bucket
-can u help why this is not seen in minio bucket with todays date it shows the all old data so deleted the emission and process input folder and again upload the new files but not seen this in minio bucket

**Comment by priya23595** [comment]
-Actually only event and dimension files are available to upload so am using new programs api to upload
-But i dont understand about what exactly we upload on national programs api so that it will create emission folder

**Comment by bommalla** [comment]
@priya23595 Using national program API need to upload all VSK six programs raw files(15 zip file).

**Comment by priya23595** [comment]
for other new use case related data also need to upload VSK data files ?

**Comment by bommalla** [comment]
If you need VSK programs you can ingest the files using national_program API. Else not needed

**Reply by priya23595** [reply]
But how to see uploaded files in minio bucket

**Comment by priya23595** [comment]
when am uploading VSK files then it will get Schema dose not exist bcz am only defining event and dimension files related to our new data
so can we connect to resolve this confusion

**Comment by bommalla** [comment]
Please find the below google drive URL for ingestion E2E video is there.

https://drive.google.com/file/d/1GLYfQMPJHWMfDseLiaZg1G46Zfu2Ie--/view

Thanks & Regards
Rajesh

On Thu, Aug 3, 2023 at 5:39 PM Priyanka Chincholi ***@***.***>
wrote:

> But how to see uploaded files in minio bucket
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/Sunbird-cQube/discussions/809#discussioncomment-6627121>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AUXQBUUKDWLPFQPKXRJMX5DXTOIF7ANCNFSM6AAAAAA2VR6Y3U>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
Regards,





*Rajesh Bommalla*

Sr.Data Engineer


*M* +91 9908089539

*E* ***@***.***

www.tibilsolutions.com
[image: LinkedIn] <https://www.linkedin.com/company/1913934> [image:
Facebook] <https://www.facebook.com/TibilSolutions> [image: Twitter]
<https://twitter.com/TibilSolutions> [image: Insta]
<https://www.instagram.com/tibilsolutions/>
*This message (including any attachments) may contain confidential,
proprietary, privileged, and/or private information. The information is
intended to be for the use of the individual or entity designated above. If
you are not the intended recipient of this message, please notify the
sender immediately, and delete the message and any attachments. Any
disclosure, reproduction, distribution, or other use of this message or any
attachments by an individual or entity other than the intended recipient is
prohibited.*

**Reply by bommalla** [reply]
https://drive.google.com/file/d/1GLYfQMPJHWMfDseLiaZg1G46Zfu2Ie--/view?usp=sharing

**Comment by priya23595** [comment]
i am able to upload custom dimension and event files to the minio bucket, however after running the scheduler the custom data is not reflected into the database. What changes do i have to make so that custom data can be reflected into the database.

**Reply by bommalla** [reply]
@priya23595 what are the programs we are ingesting ? VSK or New_Programs ? 

1. Every grammar file should have one data file then only datasets are created in the database otherwise it won't created.datasets are won't created in the database If any temp file or data file are not created in the dimension folder 
2. Need to remove the temp files and unused grammar files from the dimension folder.
3. The adaptor API will pick the files from the bucket and create one process_input folder and store the all event files with program name separately in the process input folder.
4. The "Run Latest Code local" nifi processor take input files + dimensions based on that mapping will happens later Nifi having certain datasets tables list with queries are executed and storing output to respective dataset table

**Comment by priya23595** [comment]
can we connect for this part i will show u what the costume data is and how it goes in minio bucket and what to do next for visualisation

**Comment by bommalla** [comment]
Yeah sure I'll connect.Please tell me your availability then I can
schedule call accordingly.

Thanks & Regards
Rajesh


On Wed, Aug 9, 2023 at 12:11 PM Priyanka Chincholi ***@***.***>
wrote:

> can we connect for this part i will show u what the costume data is and
> how it goes in minio bucket and what to do next for visualisation
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/Sunbird-cQube/discussions/809#discussioncomment-6676678>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AUXQBUXWWEWXA4JAM3WVYRLXUMWILANCNFSM6AAAAAA2VR6Y3U>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
Regards,





*Rajesh Bommalla*

Sr.Data Engineer


*M* +91 9908089539

*E* ***@***.***

www.tibilsolutions.com
[image: LinkedIn] <https://www.linkedin.com/company/1913934> [image:
Facebook] <https://www.facebook.com/TibilSolutions> [image: Twitter]
<https://twitter.com/TibilSolutions> [image: Insta]
<https://www.instagram.com/tibilsolutions/>
*This message (including any attachments) may contain confidential,
proprietary, privileged, and/or private information. The information is
intended to be for the use of the individual or entity designated above. If
you are not the intended recipient of this message, please notify the
sender immediately, and delete the message and any attachments. Any
disclosure, reproduction, distribution, or other use of this message or any
attachments by an individual or entity other than the intended recipient is
prohibited.*

**Comment by priya23595** [comment]
please plan according to your time and send the meet link

**Comment by bommalla** [comment]
Yeah sue.Can you please send me your email address?

Thanks & Regards
Rajesh

On Wed, Aug 9, 2023 at 12:35 PM Priyanka Chincholi ***@***.***>
wrote:

> please plan according to your time and send the meet link
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/Sunbird-cQube/discussions/809#discussioncomment-6676854>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AUXQBUR7FRK5TBR4R4IHG2TXUMZB7ANCNFSM6AAAAAA2VR6Y3U>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
Regards,





*Rajesh Bommalla*

Sr.Data Engineer


*M* +91 9908089539

*E* ***@***.***

www.tibilsolutions.com
[image: LinkedIn] <https://www.linkedin.com/company/1913934> [image:
Facebook] <https://www.facebook.com/TibilSolutions> [image: Twitter]
<https://twitter.com/TibilSolutions> [image: Insta]
<https://www.instagram.com/tibilsolutions/>
*This message (including any attachments) may contain confidential,
proprietary, privileged, and/or private information. The information is
intended to be for the use of the individual or entity designated above. If
you are not the intended recipient of this message, please notify the
sender immediately, and delete the message and any attachments. Any
disclosure, reproduction, distribution, or other use of this message or any
attachments by an individual or entity other than the intended recipient is
prohibited.*

**Comment by priya23595** [comment]
My Email_Id:
priyanka_c@tekditechnologies.com

Thank you

**Comment by priya23595** [comment]
hello
 vim natitonwideusers-dimention.grammar.csv Index,,, string,string,string,string state_id,state_name,"suborg_id",suborg_name

am creating grammar file using vim command but it not get created plzz help me to correct this command

**Comment by bommalla** [comment]
@Priyanka Chincholi ***@***.***> Please follow the
below steps.

1. First need install vim by using below command
apt-get install vim
2. vim <grammar file name>
3. copy all the grammar file content and paste it
4. Then save the file by using the below command
press esc button :wq!

Thanks & Regards
Rajesh



On Fri, Aug 11, 2023 at 4:56 PM Priyanka Chincholi ***@***.***>
wrote:

> hello
> vim natitonwideusers-dimention.grammar.csv Index,,,
> string,string,string,string state_id,state_name,"suborg_id",suborg_name
>
> am creating grammar file using vim command but it not get created plzz
> help me to correct this command
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/Sunbird-cQube/discussions/809#discussioncomment-6700692>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AUXQBUXE4FS5TAAVEFAIUCDXUYJHZANCNFSM6AAAAAA2VR6Y3U>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
Regards,





*Rajesh Bommalla*

Sr.Data Engineer


*M* +91 9908089539

*E* ***@***.***

www.tibilsolutions.com
[image: LinkedIn] <https://www.linkedin.com/company/1913934> [image:
Facebook] <https://www.facebook.com/TibilSolutions> [image: Twitter]
<https://twitter.com/TibilSolutions> [image: Insta]
<https://www.instagram.com/tibilsolutions/>
*This message (including any attachments) may contain confidential,
proprietary, privileged, and/or private information. The information is
intended to be for the use of the individual or entity designated above. If
you are not the intended recipient of this message, please notify the
sender immediately, and delete the message and any attachments. Any
disclosure, reproduction, distribution, or other use of this message or any
attachments by an individual or entity other than the intended recipient is
prohibited.*

**Comment by priya23595** [comment]
vim natitonwideusers-dimention.grammar.csv 
Index,,, 
string,string,string,string 
state_id,state_name,suborg_id,suborg_name

 natitonwideusers-dimention.grammar.csv Grammar file name and its content is this correct

**Comment by priya23595** [comment]
All Grammar files are created now am going to start ingesting the data

**Comment by priya23595** [comment]
Hello @RNarmatha , @bommalla 
Am creating the grammar files of custom data and Uploading the all dimension and event files but in the database Dataset is not generating 
 after running the scheduler APIs,Can u tell me why this is not generating please refer attachment ?
![Screenshot from 2023-08-17 12-10-53](https://github.com/Sunbird-cQube/community/assets/136596601/f28e95f0-c4ea-4282-b0ea-4c198ce21e8d)
![Screenshot from 2023-08-17 12-10-58](https://github.com/Sunbird-cQube/community/assets/136596601/01f8f068-4f8d-443e-be81-d0a1fa6b0621)
![Screenshot from 2023-08-17 12-11-40](https://github.com/Sunbird-cQube/community/assets/136596601/82b59817-0f1e-420c-bdfe-73df02edbded)

**Comment by bommalla** [comment]
@Priyanka Chincholi ***@***.***> Please check whether
the nationwide users' grammar file is properly created or not ?

Please send me the nationwide users event data.I'll check it from my end.

Thanks & Regards

On Thu, Aug 17, 2023 at 12:14 PM Priyanka Chincholi <
***@***.***> wrote:

> Hello @RNarmatha <https://github.com/RNarmatha> , @bommalla
> <https://github.com/bommalla>
> Am creating the grammar files of custom data and Uploading the all
> dimension and event files but in the database Dataset is not generating
> after running the scheduler APIs,Can u tell me why this is not generating
> please refer attachment ?
> [image: Screenshot from 2023-08-17 12-10-53]
> <https://user-images.githubusercontent.com/136596601/261220990-f28e95f0-c4ea-4282-b0ea-4c198ce21e8d.png>
> [image: Screenshot from 2023-08-17 12-10-58]
> <https://user-images.githubusercontent.com/136596601/261221022-01f8f068-4f8d-443e-be81-d0a1fa6b0621.png>
> [image: Screenshot from 2023-08-17 12-11-40]
> <https://user-images.githubusercontent.com/136596601/261221042-82b59817-0f1e-420c-bdfe-73df02edbded.png>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/Sunbird-cQube/discussions/809#discussioncomment-6747813>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AUXQBUVPYL6QDCTEHOTRIBDXVW4WRANCNFSM6AAAAAA2VR6Y3U>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Regards,





*Rajesh Bommalla*

Sr.Data Engineer


*M* +91 9908089539

*E* ***@***.***

www.tibilsolutions.com
[image: LinkedIn] <https://www.linkedin.com/company/1913934> [image:
Facebook] <https://www.facebook.com/TibilSolutions> [image: Twitter]
<https://twitter.com/TibilSolutions> [image: Insta]
<https://www.instagram.com/tibilsolutions/>
*This message (including any attachments) may contain confidential,
proprietary, privileged, and/or private information. The information is
intended to be for the use of the individual or entity designated above. If
you are not the intended recipient of this message, please notify the
sender immediately, and delete the message and any attachments. Any
disclosure, reproduction, distribution, or other use of this message or any
attachments by an individual or entity other than the intended recipient is
prohibited.*

**Comment by priya23595** [comment]
[natitonwidecontent-event-data.csv](https://github.com/Sunbird-cQube/community/files/12368348/natitonwidecontent-event-data.csv)
[natitonwideusers-event-data.csv](https://github.com/Sunbird-cQube/community/files/12368350/natitonwideusers-event-data.csv)

**Comment by priya23595** [comment]
Grammar file is Created correctly please check on your side

**Comment by bommalla** [comment]
@Priyanka Chincholi ***@***.***> Please check the all
diemasion and event data files if there is foreign key relation or not. if
not datasets are not created.

Thanks & Regards
Rajesh

On Thu, Aug 17, 2023 at 4:16 PM Priyanka Chincholi ***@***.***>
wrote:

> Grammar file is Created correctly please check on your side
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/Sunbird-cQube/discussions/809#discussioncomment-6750016>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AUXQBUVLXOKXX3RKF34JALTXVXY6XANCNFSM6AAAAAA2VR6Y3U>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Regards,





*Rajesh Bommalla*

Sr.Data Engineer


*M* +91 9908089539

*E* ***@***.***

www.tibilsolutions.com
[image: LinkedIn] <https://www.linkedin.com/company/1913934> [image:
Facebook] <https://www.facebook.com/TibilSolutions> [image: Twitter]
<https://twitter.com/TibilSolutions> [image: Insta]
<https://www.instagram.com/tibilsolutions/>
*This message (including any attachments) may contain confidential,
proprietary, privileged, and/or private information. The information is
intended to be for the use of the individual or entity designated above. If
you are not the intended recipient of this message, please notify the
sender immediately, and delete the message and any attachments. Any
disclosure, reproduction, distribution, or other use of this message or any
attachments by an individual or entity other than the intended recipient is
prohibited.*

**Comment by priya23595** [comment]
hello sir 
can we connect for the showing u the grammar files and issue while visualising the data

**Comment by bommalla** [comment]
Sure. @Priyanka Chincholi ***@***.***> I scheduled a
call at 2:30PM today.

Thanks & Regards
Rajesh


On Wed, Aug 30, 2023 at 12:30 PM Priyanka Chincholi <
***@***.***> wrote:

> hello sir
> can we connect for the showing u the grammar files and issue while
> visualising the data
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/Sunbird-cQube/discussions/809#discussioncomment-6860655>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AUXQBUUFHRRBE3PH2EGZXZDXX3QI5ANCNFSM6AAAAAA2VR6Y3U>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Regards,





*Rajesh Bommalla*

Sr.Data Engineer


*M* +91 9908089539

*E* ***@***.***

www.tibilsolutions.com
[image: LinkedIn] <https://www.linkedin.com/company/1913934> [image:
Facebook] <https://www.facebook.com/TibilSolutions> [image: Twitter]
<https://twitter.com/TibilSolutions> [image: Insta]
<https://www.instagram.com/tibilsolutions/>
*This message (including any attachments) may contain confidential,
proprietary, privileged, and/or private information. The information is
intended to be for the use of the individual or entity designated above. If
you are not the intended recipient of this message, please notify the
sender immediately, and delete the message and any attachments. Any
disclosure, reproduction, distribution, or other use of this message or any
attachments by an individual or entity other than the intended recipient is
prohibited.*

**Comment by priya23595** [comment]
ok sir

**Comment by priya23595** [comment]
can we connect at 3 because am having meeting at 2.30

**Comment by RNarmatha** [comment]
Hi Priyanka,
Sure We will connect at 3 pm

Thanks
Narmatha
